### ISSUE TITLE
aria-describedby Telephone component does not announce description

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/vets-design-system-documentation#2254]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

We are currently using the `<va-telephone>` component with `aria-describedby` in the Facility Locator. The purpose is to clearly label each phone number for each facility so a user understands which number to use for which purpose. However, when tabbing through the page using a screen reader, the label is not being read out with the phone number. For example, I am only hearing "617 248 1000" instead of "617 248 1000 Main Number"

## What I expected to happen

I would expect, each phone number to be read followed by the phone number label, i.e. Main number, Mental Health number, etc.

## Reproducing
- Formation version:
- Device: NVDA/Windows, JAWS/Windows, VoiceOver/Mac
- Browser: (e.g. Firefox, IE 11)

Steps to reproduce:

1. Go to [the Facility Locator](https://www.va.gov/find-locations)
2. Search for a VA health Center center
3. Turn on a screen reader
4. Start tabbing through the list of results
5. Notice that when you tab to a facility phone number, there is no indication of what type of phone number it is, no description is read.

## Urgency
How urgent is this request? Please select the approriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [x] Other

**Comment by coforma-jamie** [comment]
Verifying that this is an issue.

An example of this working properly is `<va-text-input>`, where it adds its optional `message-aria-describedby` to the input itself:
![image](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/135633989/8a2380b5-75b3-4de5-8448-ee54c7b4ebd5)

In contrast, `<va-telephone>` adds its optional `aria-describedby` to the COMPONENT, not the link in the shadow DOM:
![image](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/assets/135633989/83fe4d1f-a46b-4cc3-b8c6-b507a994e8f5)

`<va-telephone>` needs to be fixed so that `aria-describedby` is added to `<a>` in the rendered code.

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2254?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc
